src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -1003,12 +1003,12 @@ module SimulationCode =
             | Null -> []
             | Value location -> [
                 // since the line numbers throughout the generated code are 1-based, let's also choose them 1-based here
```

Comment:
Why are line numbers in the generated code 1-based? And if they are, should we adopt the a similar strategy to what you did in the compiler, to use a zero-based system throughout the logic and convert at the very end?

Reply:
I think it's referring to the [`#line`](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives/preprocessor-line) markers added to the generated code.

The `LineNumber` property that we define should probably be zero-based, and we can add 1 when the `#line` marker is actually generated, but it looks like the code is doing something weird where when `LineNumber` is `Some 0`, `#line hidden` is generated instead of an actual line number.

I'm a little afraid to change that and would prefer to keep the main refactoring on the compiler repo - the changes here were just the bare minimum needed to adapt to the compiler changes.